[Auth] Can't sign in

Clicking "Sign in with Google" on sign in page and selecting an account has no effect.

The issue appears to be in `Navigator.showSignInScreen`; the id returned by `getNavController().getCurrentDestination().getId()` is not recognized.

Stack trace:
```
2020-01-05 23:22:48.893 31109-31109/com.google.android.gnd E/RxDebug: Unhandled Rx error
    io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.IllegalArgumentException: 2131296541 id not found
        at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
        at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
        at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
        at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:67)
        at com.uber.autodispose.HalfSerializer.onNext(HalfSerializer.java:121)
        at com.uber.autodispose.AutoDisposingObserverImpl.onNext(AutoDisposingObserverImpl.java:76)
        at io.reactivex.internal.util.NotificationLite.accept(NotificationLite.java:246)
        at io.reactivex.subjects.BehaviorSubject$BehaviorDisposable.test(BehaviorSubject.java:569)
        at io.reactivex.subjects.BehaviorSubject$BehaviorDisposable.emitNext(BehaviorSubject.java:564)
        at io.reactivex.subjects.BehaviorSubject.onNext(BehaviorSubject.java:268)
        at com.google.android.gnd.system.AuthenticationManager.onActivityResult(AuthenticationManager.java:123)
        at com.google.android.gnd.system.AuthenticationManager.lambda$vqw1QPZqSWMLmiSZ3r0w--0ct1w(AuthenticationManager.java)
        at com.google.android.gnd.system.-$$Lambda$AuthenticationManager$vqw1QPZqSWMLmiSZ3r0w--0ct1w.accept(lambda)
        at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
        at com.akaita.java.rxjava2debug.extensions.ObservableOnAssembly$OnAssemblyObserver.onNext(ObservableOnAssembly.java:55)
        at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:52)
        at io.reactivex.subjects.PublishSubject$PublishDisposable.onNext(PublishSubject.java:308)
        at io.reactivex.subjects.PublishSubject.onNext(PublishSubject.java:228)
        at com.google.android.gnd.system.ActivityStreams.onActivityResult(ActivityStreams.java:53)
        at com.google.android.gnd.MainActivity.onActivityResult(MainActivity.java:192)
        at android.app.Activity.dispatchActivityResult(Activity.java:6932)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4085)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4132)
        at android.app.ActivityThread.-wrap20(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1533)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
     Caused by: java.lang.IllegalArgumentException: 2131296541 id not found
        at com.google.android.gnd.ui.common.Navigator.showSignInScreen(Navigator.java:126)
        at com.google.android.gnd.MainViewModel.onSignedOut(MainViewModel.java:54)
        at com.google.android.gnd.MainActivity.onSignInError(MainActivity.java:139)
        at com.google.android.gnd.MainActivity.onSignInStateChange(MainActivity.java:131)
        at com.google.android.gnd.MainActivity.lambda$RvpWYbC44xMC5rQnCDl7ZoI8B8A(MainActivity.java)
        at com.google.android.gnd.-$$Lambda$MainActivity$RvpWYbC44xMC5rQnCDl7ZoI8B8A.accept(lambda)
        at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
        at com.uber.autodispose.HalfSerializer.onNext(HalfSerializer.java:121) 
        at com.uber.autodispose.AutoDisposingObserverImpl.onNext(AutoDisposingObserverImpl.java:76) 
        at io.reactivex.internal.util.NotificationLite.accept(NotificationLite.java:246) 
        at io.reactivex.subjects.BehaviorSubject$BehaviorDisposable.test(BehaviorSubject.java:569) 
        at io.reactivex.subjects.BehaviorSubject$BehaviorDisposable.emitNext(BehaviorSubject.java:564) 
        at io.reactivex.subjects.BehaviorSubject.onNext(BehaviorSubject.java:268) 
        at com.google.android.gnd.system.AuthenticationManager.onActivityResult(AuthenticationManager.java:123) 
        at com.google.android.gnd.system.AuthenticationManager.lambda$vqw1QPZqSWMLmiSZ3r0w--0ct1w(AuthenticationManager.java) 
        at com.google.android.gnd.system.-$$Lambda$AuthenticationManager$vqw1QPZqSWMLmiSZ3r0w--0ct1w.accept(lambda) 
        at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63) 
        at com.akaita.java.rxjava2debug.extensions.ObservableOnAssembly$OnAssemblyObserver.onNext(ObservableOnAssembly.java:55) 
        at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:52) 
        at io.reactivex.subjects.PublishSubject$PublishDisposable.onNext(PublishSubject.java:308) 
        at io.reactivex.subjects.PublishSubject.onNext(PublishSubject.java:228) 
        at com.google.android.gnd.system.ActivityStreams.onActivityResult(ActivityStreams.java:53) 
        at com.google.android.gnd.MainActivity.onActivityResult(MainActivity.java:192) 
        at android.app.Activity.dispatchActivityResult(Activity.java:6932) 
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4085) 
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4132) 
        at android.app.ActivityThread.-wrap20(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1533) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6119) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) 
```

--- a/gnd/src/main/java/com/google/android/gnd/MainActivity.java
+++ b/gnd/src/main/java/com/google/android/gnd/MainActivity.java
@@ -231,10 +231,10 @@ public class MainActivity extends DaggerAppCompatActivity {
 
   private int getCurrentNavDestinationId() {
     NavDestination destination = getNavController().getCurrentDestination();
-    if (destination != null) {
-      return destination.getId();
+    if (destination == null) {
+      return -1;
     }
-    return -1;
+    return destination.getId();
   }
 
   private void onToolbarUpClicked() {

--- a/gnd/src/main/java/com/google/android/gnd/system/AuthenticationManager.java
+++ b/gnd/src/main/java/com/google/android/gnd/system/AuthenticationManager.java
@@ -119,7 +119,7 @@ public class AuthenticationManager {
           GoogleSignIn.getSignedInAccountFromIntent(activityResult.getData());
       onGoogleSignIn(googleSignInTask.getResult(ApiException.class));
     } catch (ApiException e) {
-      Log.w(TAG, "Sign in failed, GoogleSignInStatusCodes:  " + e.getStatusCode());
+      Log.w(TAG, "Sign in failed: " + e);
       signInState.onNext(new SignInState(e));
     }
   }

--- a/gnd/src/main/java/com/google/android/gnd/ui/common/Navigator.java
+++ b/gnd/src/main/java/com/google/android/gnd/ui/common/Navigator.java
@@ -122,6 +122,9 @@ public class Navigator {
       case R.id.home_screen_fragment:
         navigate(HomeScreenFragmentDirections.fromHomeScreenToSignInScreen());
         break;
+      case R.id.sign_in_fragment:
+        // Sign in screen already active.
+        break;
       default:
         throw new IllegalArgumentException(currentNavDestinationId + " id not found");
     }

